src/parser/mod.rs

```diff
@@ -9766,8 +9833,12 @@ impl<'a> Parser<'a> {
                     Ok(DataType::AnyType)
                 }
                 Keyword::TABLE => {
-                    let columns = self.parse_returns_table_columns()?;
-                    Ok(DataType::Table(columns))
+                    if self.peek_token() != Token::LParen {
+                        Ok(DataType::Table(None))
```

Comment:
can we add a comment here explaining what the LParen check is for?

Reply:
Done, and I simplified the if/else 👍